plugins/dig-now.cpp

```diff
@@ -817,12 +810,25 @@ command_result dig_now(color_ostream &out, std::vector<std::string> &params) {
 
     do_dig(out, dug_coords, item_coords, options);
     create_boulders(out, item_coords, options);
-    post_process_dug_tiles (out, dug_coords);
+    post_process_dug_tiles(out, dug_coords);
 
     // force the game to recompute its walkability cache
     world->reindex_pathfinding = true;
 
-    return CR_OK;
+    return true;
+}
+
+command_result dig_now(color_ostream &out, std::vector<std::string> &params) {
+    CoreSuspender suspend;
+
+    dig_now_options options;
+    if (!get_options(out, options, params) || options.help)
+    {
+        print_help(out);
+        return options.help ? CR_OK : CR_FAILURE;
+    }
+
+    return dig_now_impl(out, options) ? CR_OK : CR_FAILURE;
 }
```

Comment:
Just not sure about stuff like this related to `dig-now` as a plugin

Reply:
what do you mean?

Reply:
On the subject of linking against your symbol definitions.
```dfhack_plugin(channel-safely channel-safely.h channel-safely.cpp channel-safely-plugin.cpp dig-now.h dig-now.cpp)```
pretty sure those .h files need not be in that cmake line.. but added them cause I don't know one way or the other

Reply:
Oh, I thought you'd be dynamically linking via LINK_LIBRARIES (which wouldn't have duplicated symbol definitions, but still would not be good to do since you might *add* top level plugin symbol definitions that you don't want.) But anyway, with the latest code you don't need to add any dig-now files to your plugin directly. You just need to link in Lua.

Reply:
Oh, and the .h files probably don't need to be there. They're not listed for any other plugin that has .h files..